tests/test_concentration.py

```diff
@@ -18,16 +18,34 @@ def test_duffy(mass, mdef):
     assert np.allclose(duffy.cm(mass), duffyc.cm(mass))
 
 
-def test_ludlow_vs_colossus():
+@pytest.mark.parametrize("lu16", [cm.Ludlow2016, cm.Ludlow2016Empirical])
+def test_ludlow_vs_colossus(lu16):
     """Test the Ludlow relation between native and colossus implementations."""
     mf = MassFunction()
 
     L16Colossus = cm.make_colossus_cm(model="ludlow16")
 
-    l16 = cm.Ludlow16(filter0=mf.normalised_filter)
+    l16 = lu16(filter0=mf.normalised_filter)
     l16c = L16Colossus(filter0=mf.normalised_filter)
 
     m = np.logspace(10, 15, 100)
 
     # TODO: for masses of ~1e10, halomod gets c(m) = 16, while colossus gets ~12. Need to fix.
     assert np.allclose(l16.cm(m), l16c.cm(m), rtol=0.3)
+
+
+def test_lud16em_warning():
+    mf = MassFunction()
+    l16 = cm.Ludlow2016Empirical(filter0=mf.normalised_filter)
+    m = np.logspace(11, 15, 100)
+    with pytest.warns(UserWarning):
```

Comment:
Read this: https://docs.pytest.org/en/stable/warnings.html#ensuring-code-triggers-a-deprecation-warning

Reply:
Hey @zhaotingchen did this help?

Reply:
Oh sorry I forgot to mention that. So it's not the problem with testing this warning, it's that I can't get the code to trigger this warning. But somehow currently these lines are getting covered so I can't be bothered with why.